src/nvim/popupmenu.c

```diff
@@ -710,6 +773,55 @@ void pum_redraw(void)
             xfree(st);
             grid_col -= width;
           } else {
+            int size = (int)strlen(st);
+            int cells = (int)mb_string2cells(st);
+            need_ellipsis = p_pmw > ellipsis_width && pum_width == p_pmw
+                            && grid_col + cells > pum_col + pum_width;
```

Comment:
This condition is wrong as it doesn't handle multigrid. `pum_col` should be changed to `col_off`. Please see #33114.

Reply:
aha duplicate again :)